includes/class-wc-taxjar-transaction-sync.php

```diff
@@ -620,40 +625,41 @@ public function get_orders_to_backfill( $start_date = null, $end_date = null, $f
 		$valid_post_statuses = apply_filters( 'taxjar_valid_post_statuses_for_sync', array( 'wc-completed', 'wc-refunded' ) );
 		$post_status_string = "( '" . implode( "', '", $valid_post_statuses ) . " ')";
 
+		$should_validate_completed_date = WC_Taxjar_Transaction_Sync::should_validate_order_completed_date();
+
 		if ( $force ) {
-			$posts = $wpdb->get_results(
-					"
-				SELECT p.id 
-				FROM {$wpdb->posts} AS p 
-				INNER JOIN {$wpdb->postmeta} AS order_meta_completed_date ON ( p.id = order_meta_completed_date.post_id )  AND ( order_meta_completed_date.meta_key = '_completed_date' ) 
-				WHERE p.post_type = 'shop_order' 
-				AND p.post_status IN {$post_status_string} 
-				AND p.post_date >= '{$start_date}' 
-				AND p.post_date < '{$end_date}' 
-				AND order_meta_completed_date.meta_value IS NOT NULL 
-				AND order_meta_completed_date.meta_value != '' 
-				ORDER BY p.post_date ASC
-				", ARRAY_N
-			);
+			$query = "SELECT p.id FROM {$wpdb->posts} AS p ";
+
+			if ( $should_validate_completed_date ) {
+				$query .= "INNER JOIN {$wpdb->postmeta} AS order_meta_completed_date ON ( p.id = order_meta_completed_date.post_id )  AND ( order_meta_completed_date.meta_key = '_completed_date' ) ";
```

Comment:
Seeing an additional space here on lines 634, 648, 651 between `)` and `AND`: `)  AND (`

Reply:
Good catch, fixed in the most recent commit.